packages/calcite-components/src/tests/utils.ts

```diff
@@ -426,51 +427,68 @@ export function toBeNumber(): any {
   };
 }
 
+type HTMLSelectableElement = HTMLElement & { selectedItems: HTMLElement[] };
+
 interface SelectedItemsAssertionOptions {
   /**
    * IDs from items to assert selection
    */
   expectedItemIds: string[];
 }
 
+type SelectionEventTestWindow = GlobalTestProps<{ eventDetail: Selection }>;
+
+interface SelectedItemsAsserter {
+  (expectedItemIds: SelectedItemsAssertionOptions["expectedItemIds"]): Promise<void>;
+}
+
 /**
- * Test helper for selected items. Expects items to have IDs to test against.
+ * Creates a selected items asserter for a selectable component.
+ *
+ * @example
+ *
+ * const page = await newE2EPage();
+ * await page.setContent(
+ *   html`<calcite-dropdown open>
+ *     <calcite-button id="trigger" slot="trigger">Open dropdown</calcite-button>
+ *     <calcite-dropdown-group id="group-1" selection-mode="single">
+ *       <calcite-dropdown-item id="item-1"> Dropdown Item Content </calcite-dropdown-item>
+ *       <calcite-dropdown-item id="item-2" selected> Dropdown Item Content </calcite-dropdown-item>
+ *       <calcite-dropdown-item id="item-3"> Dropdown Item Content </calcite-dropdown-item>
+ *     </calcite-dropdown-group>
+ *   </calcite-dropdown>`,
+ * );
  *
- * Note: assertSelectedItems.setUpEvents must be called before using this method
+ * const assertSelectedItems = await createSelectedItemsAsserter(page, "calcite-dropdown", "calciteDropdownSelect");
+ * await page.click("#item-2");
+ * await assertSelectedItems({ expectedItemIds: ["item-2"] });
  *
- * @param groupElementTagName - The custom element tag name
- * @param page - The E2EPage instance
- * @param SelectedItemsAssertionOptions - Object containing selectable test assertion options
- * @param SelectedItemsAssertionOptions.expectedItemIds - An array of ids to expect to be selected from an interaction
+ * @param page - the e2e page
+ * @param selectableComponentTagName - the tag name of the selectable group element
+ * @param selectionEventName - the name of the selection event
  */
-export async function assertSelectedItems(
-  groupElementTagName: string,
+export async function createSelectedItemsAsserter(
   page: E2EPage,
-  { expectedItemIds }: SelectedItemsAssertionOptions,
-): Promise<void> {
-  await page.waitForTimeout(100);
```

Comment:
I never did understand what this waitForTimeout was for, so its cool to see that removed.